crates/core/src/trap.rs

```diff
@@ -216,7 +216,7 @@ impl StdError for Trap {
 /// See [`Trap`] for details.
 ///
 /// [`Trap`]: struct.Trap.html
-#[derive(Debug, Copy, Clone)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
```

Comment:
Could you cherry pick this line to master firstly?

Reply:
No, this PR is WIP for good reason. Many, if not everything here is subject to change. We can talk about this once the PR is no longer considered WIP.